vyper/functions/functions.py

```diff
@@ -679,54 +679,30 @@ def raw_call(expr, args, kwargs, context):
         location='memory',
     )
 
+    # build LLL for call or delegatecall
     if delegate_call.value == 1:
-        z = LLLnode.from_list(
-            [
-                'seq',
-                copier,
-                [
-                    'assert',
-                    [
-                        'delegatecall',
-                        gas,
-                        to,
-                        ['add', placeholder_node, 32],
-                        ['mload', placeholder_node],
-                        ['add', output_node, 32],
-                        outsize,
-                    ],
-                ],
-                ['mstore', output_node, outsize],
-                output_node,
-            ],
-            typ=ByteArrayType(outsize),
-            location='memory',
-            pos=getpos(expr),
-        )
+        call_lll = ['delegatecall', gas, to]
     else:
-        z = LLLnode.from_list(
-            [
-                'seq',
-                copier,
-                [
-                    'assert',
-                    [
-                        'call',
-                        gas,
-                        to,
-                        value,
-                        ['add', placeholder_node, 32],
-                        ['mload', placeholder_node],
-                        ['add', output_node, 32],
-                        outsize,
-                    ],
-                ],
-                ['mstore', output_node, outsize],
-                output_node,
-            ],
-            typ=ByteArrayType(outsize), location='memory', pos=getpos(expr)
-        )
-    return z
+        call_lll = ['call', gas, to, value]
+    call_lll += [
+        ['add', placeholder_node, 32],
+        ['mload', placeholder_node],
+        # if there is no return value, the return offset can be 0
+        ['add', output_node, 32] if outsize else 0,
+        outsize
+    ]
+
+    # build sequence LLL
+    seq = ['seq', copier, ['assert', call_lll]]
+    if outsize:
+        # only copy the return value to memory if outsize > 0
+        seq += [['mstore', output_node, outsize], output_node]
```

Comment:
This whole section (686-699) is sort of ugly with all the mutations. Is there a way to build all the parts separately and combine them without mutation?